src/cdk/datetime/datepicker.ts

```diff
@@ -0,0 +1,131 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {
+  ChangeDetectionStrategy,
+  Component,
+  Input,
+  Optional,
+  ViewEncapsulation,
+  OnDestroy,
+  ContentChild,
+} from '@angular/core';
+import {Subject, Subscription} from 'rxjs';
+import {DateAdapter} from './date-adapter';
+import {CdkDatepickerInput} from './datepicker-input';
+import {CalendarView} from './calendar-view';
+
+/** Used to generate a unique ID for each datepicker instance. */
+let datepickerUid = 0;
+
+// TODO(vwei): We use a component instead of a directive here so the user can use implicit
+// template reference variables (e.g. #d vs #d="cdkDatepicker"). We can change this to a directive
+// if angular adds support for `exportAs: '$implicit'` on directives.
+/** Component responsible for managing the datepicker CDK. */
+@Component({
+  moduleId: module.id,
+  selector: 'cdk-datepicker',
+  template: '<ng-content></ng-content>',
+  exportAs: 'cdkDatepicker',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  encapsulation: ViewEncapsulation.None,
+})
+export class CdkDatepicker<D> implements OnDestroy {
+  /** The initial date of the datepicker CDK. */
+  @Input()
+  get startAt(): D | null {
+    // If an explicit startAt is set we start there, otherwise we start at whatever the currently
+    // selected value is.
+    return this._startAt || (this._datepickerInput ? this._datepickerInput.value : null);
+  }
+  set startAt(value: D | null) {
+    this._startAt = this._getValidDateOrNull(this._dateAdapter.deserialize(value));
+  }
+  private _startAt: D | null;
+
+  /** The child of the datepicker */
+  @ContentChild(CalendarView) view: CalendarView<D>;
+
```

Comment:
I think we still want the `@Input() disabled`, all calendars should have this though the implementation may be different. For ones where the calendar is always shown maybe they just disable clicking on any of the dates or something